### DISCUSSION TITLE
Textbox function not working properly

### DISCUSSION THREAD

**Discussion by JohnnyFruitbasket** [TomSchimansky/CustomTkinter#2316] [Q&A]
Trying to implement a function that creates and displays a textbox. This is the function specific code so far:
```python
def display_query_result(self, result_text):
        if not hasattr(self, "result_textbox"):
            self.result_textbox = ctk.CTkTextbox(self, height=100, width=50, fg_color="white", text_color="black")
            self.result_textbox.grid(row=12, column=0, padx=10, pady=10, sticky="ew", columnspan=3)
        else:
                self.result_textbox.configure(state="normal")
                self.result_textbox.delete("0.0", "end")
        self.result_textbox.insert("0.0", result_text)  
        self.result_textbox.configure(state="disabled") 
```
While the first call of this function works just fine and it displays the textbox with the query in the app, whenever I try to call the function again it does not delete the text and seems unable to generate a new result. I tested it repeatedly and it loops fine through the function, the second time I call it in the app, it accesses the "else" part. My guess was that the delete method was unable to access the textbox, so I tried the configure method, but honestly, this was more of a wild guess. Anybody, any suggestion?

**Comment by dipeshSam** [comment]
@JohnnyFruitbasket 

``` Python
self.result_textbox = ctk.CTkTextbox(self, height=100, width=50, fg_color="white", text_color="black")
self.result_textbox.grid(row=12, column=0, padx=10, pady=10, sticky="ew", columnspan=3)
 ```

This part should be defined outside of the function.
To hide the text box you can use `.grid_forget()` function, and make it visible again you need to use `.grid()` again.

Others configurations are ramain same.

Hope, it may work.
Regards.

**Reply by JohnnyFruitbasket** [reply]
Thanks for your Input, I will try that
